src/main/java/ooo/oshi/foreign/windows/Kernel32Library.java

```diff
@@ -60,5 +66,29 @@ public static int getCurrentProcessId() {
 
     private static final MethodHandle getCurrentProcessId = methodHandle("GetCurrentProcessId",
             FunctionDescriptor.of(JAVA_INT));
+    
+    /**
+     * Retrieves the NetBIOS name of the local computer. This name is established at system startup, when the system
+     * reads it from the registry.
+     *
+     * @return the NetBIOS name of the local computer.
+     */
+    public static String getComputerName() {
+    	try {
+        	SegmentAllocator allocator = SegmentAllocator.implicitAllocator();
+    		Addressable buffer = allocator.allocate(WinNT.MAX_COMPUTERNAME_LENGTH + 1);
+    		Addressable size = allocator.allocate(JAVA_LONG, WinNT.MAX_COMPUTERNAME_LENGTH + 1);
+            if(!(boolean) getComputerName.invokeExact(buffer, size))	{
+            	throw new RuntimeException("Failed to get Computer Name");
+            }
+            return ((MemorySegment)buffer).getUtf8String(0);
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static final MethodHandle getComputerName = methodHandle("GetComputerNameA",
```

Comment:
You should be mapping the W functions (Unicode) rather than the A functions (ANSI / ASCII) which really don't apply for any modern version of Windows.  That also may mean allocating 2x the memory.